pallets/pool-system/src/impls.rs

```diff
@@ -394,3 +394,102 @@ impl<T: Config> InvestmentAccountant<T::AccountId> for Pallet<T> {
 		T::Tokens::burn_from(id.into(), seller, amount).map(|_| ())
 	}
 }
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarks_utils {
+	use cfg_traits::{Investment, PoolBenchmarkHelper};
+	use cfg_types::tokens::CurrencyId;
+	use frame_benchmarking::account;
+	use frame_support::traits::Currency;
+	use frame_system::RawOrigin;
+	use sp_std::vec;
+
+	use super::*;
+	use crate::tranches::TrancheMetadata;
+
+	impl<T: Config<CurrencyId = CurrencyId>> PoolBenchmarkHelper for Pallet<T>
+	where
+		T::Investments: Investment<T::AccountId, InvestmentId = T::TrancheCurrency>,
+		<T::Investments as Investment<T::AccountId>>::Amount: From<u32>,
+	{
+		type AccountId = T::AccountId;
+		type Balance = T::Balance;
+		type PoolId = T::PoolId;
+
+		fn benchmark_create_pool(pool_id: T::PoolId, admin: &T::AccountId) {
+			const FUNDS: u32 = u32::max_value();
+
+			T::Currency::make_free_balance_be(admin, T::PoolDeposit::get());
+			// Pool creation
+			Pallet::<T>::create(
+				admin.clone(),
+				admin.clone(),
+				pool_id,
+				vec![
+					TrancheInput {
+						tranche_type: TrancheType::Residual,
+						seniority: None,
+						metadata: TrancheMetadata {
+							token_name: BoundedVec::default(),
+							token_symbol: BoundedVec::default(),
+						},
+					},
+					TrancheInput {
+						tranche_type: TrancheType::NonResidual {
+							interest_rate_per_sec: One::one(),
+							min_risk_buffer: Perquintill::from_percent(10),
+						},
+						seniority: None,
+						metadata: TrancheMetadata {
+							token_name: BoundedVec::default(),
+							token_symbol: BoundedVec::default(),
+						},
+					},
+				],
+				CurrencyId::AUSD,
+				FUNDS.into(),
+				None,
+			)
+			.unwrap();
+
+			// Investment in pool
+			let investor = account::<T::AccountId>("investor_benchmark_pool", 0, 0);
+			let tranche = Pallet::<T>::pool(pool_id)
+				.unwrap()
+				.tranches
+				.tranche_id(TrancheLoc::Index(0))
+				.unwrap();
+
+			T::Permission::add(
+				PermissionScope::Pool(pool_id),
+				investor.clone(),
+				Role::PoolRole(PoolRole::TrancheInvestor(tranche, u64::MAX)),
+			)
+			.unwrap();
+
+			T::Tokens::mint_into(CurrencyId::AUSD, &investor, FUNDS.into()).unwrap();
+			T::Investments::update_investment(
+				&investor,
```

Comment:
I would at least think about it a bit before merging this like this. I agree we need benchmarking utilities and those should be abstracted via traits. But this impl here makes so many assumptions. 

We should probably take a little time to think whether this is sufficient for all benchmarks that depends on pools or whether we just create more technical debt here.

Reply:
If agree, I reference this conversation into #1237 to track this technical debt